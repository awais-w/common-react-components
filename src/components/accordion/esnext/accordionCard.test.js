import React from 'react';
import { shallow } from 'enzyme';
import AccordionCard from './accordionCard';

const defaultProps = {
  allowMultipleOpen: true,
  indx: 0,
  cardData: {
    header: {
      icon: 'forum',
      title1: '13 Jan 2020, 00:32:21',
      title2: 'Order Number: 6000002600',
      actions: [
        { icon: 'icon-url', isCustom: true, color: 'blue', tooltip: 'copy url' },
        { icon: 'zoom_out_map', color: 'blue', tooltip: 'expand transcript' },
      ],
    },
    body: 'This is card content.',
  },
  isExpanded: false,
  onCardClick: jest.fn(),
};

const setup = ({ props }) => {
  const component = shallow(<AccordionCard {...props} />);

  return {
    wrapper: component,
  };
};

describe('<AccordionCard />', () => {
  describe('with default props', () => {
    const component = setup({ props: defaultProps });

    it('handleClick', async () => {
      await component.wrapper.instance().handleClick();
      expect(component.wrapper.state().isExpanded).toEqual(true);
    });

    it('showLoadContent', async () => {
      const newComponent = setup({ props: defaultProps });
      const shouldLoadcontent = await newComponent.wrapper.instance().shouldLoadcontent();
      expect(shouldLoadcontent).toEqual(false);
    });

    it('componentDidUpdate', async () => {
      const oldProps = {
        ...defaultProps,
        isExpanded: true,
      };
      await component.wrapper.instance().componentDidUpdate(oldProps);
      expect(component.wrapper.state().isExpanded).toEqual(false);
    });

    it('should match the default snapshot', () => {
      expect(component.wrapper).toMatchSnapshot();
    });

    it('exists', () => expect(component.wrapper).toExist());
  });
});
