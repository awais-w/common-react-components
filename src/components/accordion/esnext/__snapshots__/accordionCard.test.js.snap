// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionCard /> with default props should match the default snapshot 1`] = `
<Fragment>
  <Card
    icon="forum"
    isExpanded={false}
    isLoading={false}
    onToggle={[Function]}
    rightIcons={
      Array [
        Object {
          "color": "blue",
          "icon": "icon-url",
          "isCustom": true,
          "tooltip": "copy url",
        },
        Object {
          "color": "blue",
          "icon": "zoom_out_map",
          "tooltip": "expand transcript",
        },
        Object {
          "icon": "expand_more",
          "tooltip": "",
        },
      ]
    }
    title1="13 Jan 2020, 00:32:21"
    title2="Order Number: 6000002600"
  >
    This is card content.
  </Card>
</Fragment>
`;
